### PR TITLE
Python integration sample

### PR DESCRIPTION
**Pull request by tmartin-gh**
Satisfy #810

### CODE REVIEW

File: examples/python_integration_sample/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  set(CMAKE_CUDA_ARCHITECTURES 70 CACHE STRING "Select compile target CUDA Compute Capabilities")
+endif()
+
+if(NOT DEFINED MATX_FETCH_REMOTE)
+  message(WARNING "MATX_FETCH_REMOTE not defined.  Defaulting to OFF, will use local MatX repo")
+  set(MATX_FETCH_REMOTE OFF CACHE BOOL "Set MatX repo fetch location")
+endif()
+
+project(SAMPLE_MATX_PYTHON LANGUAGES CUDA CXX)
+find_package(CUDAToolkit 12.6 REQUIRED)
```
Comment (cliffburdick) [inline examples/python_integration_sample/CMakeLists.txt:23]: Maybe decrease the one required to 12.2?

File: examples/python_integration_sample/example_matxutil.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+print(f"  GPU mempool used bytes {gpu_mempool.used_bytes()}")
+print(f"  Pinned mempool n_free_blocks {pinned_mempool.n_free_blocks()}")
+print()
+
+print("Demonstrate python to C++ to python to C++ calling chain (uses mypythonlib.py):")
+# This function calls back into python and executes a from_dlpack, consuming the dlp
+matxutil.call_python_example(dlp)
+assert(matxutil.check_dlpack_status(dlp) != 0)
+
+# Other things to try
```
Comment (cliffburdick) [inline examples/python_integration_sample/example_matxutil.py:56]: Does this comment need to be here?

**Review by cliffburdick — APPROVED** [review]

File: examples/python_integration_sample/CMakeLists.txt (hunk 1 above)
Comment (tmartin-gh) [reply examples/python_integration_sample/CMakeLists.txt]: done

File: examples/python_integration_sample/example_matxutil.py (hunk 2 above)
Comment (tmartin-gh) [reply examples/python_integration_sample/example_matxutil.py]: Removed

**Pull request merged**